Sources/Web3Core/KeystoreManager/EtherscanTransactionChecker.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  EtherscanTransactionChecker.swift
+//  Created by albertopeam on 28/12/22.
+//
+
+import Foundation
+
+public struct EtherscanTransactionChecker: TransactionChecker {
+    private let urlSession: URLSessionProxy
+    private let apiKey: String
+
+    public init(urlSession: URLSession, apiKey: String) {
+        self.urlSession = URLSessionProxyImplementation(urlSession: urlSession)
+        self.apiKey = apiKey
+    }
+
+    internal init(urlSession: URLSessionProxy, apiKey: String) {
+        self.urlSession = urlSession
+        self.apiKey = apiKey
+    }
+
+    public func hasTransactions(address: String) async throws -> Bool {
+        let urlString = "https://api.etherscan.io/api?module=account&action=txlist&address=\(address)&startblock=0&page=1&offset=1&sort=asc&apikey=\(apiKey)"
+        guard let url = URL(string: urlString) else {
+            throw EtherscanTransactionCheckerError.invalidUrl(url: urlString)
+        }
+        let request = URLRequest(url: url)
+        let result = try await urlSession.data(for: request)
+        let response = try JSONDecoder().decode(Response.self, from: result.0)
+        return !response.result.isEmpty
+    }
+}
+
+extension EtherscanTransactionChecker {
+    struct Response: Codable {
+        let result: [Transaction]
+    }
+    struct Transaction: Codable {}
```

Comment:
Could you add at least a field like `hash` to the `Transaction` struct?
Or maybe a swift doc to that struct explaining why it's empty to make sure we won't spend time in the future thinking too much about why it's empty. 🙂

I guess it is something like `We are not interested in the contents of the transaction object but rather the availability of the transaction per see for the sake of counting the number of transactions for a specific account.`.

Reply:
Apologies, I had no time yesterday.